lib/eslint.js

```diff
@@ -935,6 +935,37 @@ module.exports = (function() {
     };
 
     /**
+     * Record that a particular variable has been used in code
+     * @param {string} name The name of the variable to mark as used
+     * @returns {void}
+     */
+    api.markVariableAsUsed = function(name) {
+        var scope = this.getScope(),
+            variables,
+            i,
+            len;
+
+        do {
+            variables = scope.variables;
+            for (i = 0, len = variables.length; i < len; i++) {
+                if (variables[i].name === name) {
+                    variables[i].eslintNodeHasBeenUsed = true;
+                    return;
+                }
+            }
+        } while ( (scope = scope.upper) );
+    };
+
+    /**
+     * Check whether a variable node was previously marked as used
+     * @param {ASTNode} node The variable node to check
+     * @returns {Boolean} true if node was marked used
+     */
+    api.isMarkedAsUsed = function(node) {
+        return node.eslintNodeHasBeenUsed === true;
```

Comment:
Not sure this is quite right. Typically, function pairs accept the same type of arguments, so I'd expect this to accept a string. This is also not an accurate method name because a variable can be used without having called `markVariableAsUsed`. I think it might be best to remove this method altogether and just have `no-unused-vars` check the property directly.